commands/list/nodepools/command.go

```diff
@@ -0,0 +1,206 @@
+	AZ:             Availability zone letters used by the node pool, separated by comma
+	INSTANCE TYPE:  EC2 instance type used for worker nodes
+	NODES MIN/MAX:  The minimum and maximum number of worker nodes in this pool
+	NODES DESIRED:  Current desired number of nodes as determined by the autoscaler
+	NODES READY:    Number of nodes that are in the Ready state in kubernetes
+	CPUS:           Sum of CPU cores in nodes that are in state Ready
+	RAM (GB):       Sum of memory in GB of all nodes that are in state Ready
+
+To see all available details for a cluster, use 'gsctl show nodepool <cluster-id>/<nodepool-id>'.
+
+To list all clusters you have access to, use 'gsctl list clusters'.`,
+		PreRun: printValidation,
+		Run:    printResult,
+	}
+)
+
+const activityName = "list-nodepools"
+
+type arguments struct {
+	apiEndpoint string
+	authToken   string
+	scheme      string
+	clusterID   string
+}
+
+// defaultArgs creates arguments based on command line flags and config.
+func defaultArgs(cmdLineArgs []string) arguments {
+	endpoint := config.Config.ChooseEndpoint(flags.CmdAPIEndpoint)
+	token := config.Config.ChooseToken(endpoint, flags.CmdToken)
+	scheme := config.Config.ChooseScheme(endpoint, flags.CmdToken)
+
+	return arguments{
+		apiEndpoint: endpoint,
+		authToken:   token,
+		scheme:      scheme,
+		clusterID:   cmdLineArgs[0],
+	}
+}
+
+func verifyPreconditions(args arguments, positionalArgs []string) error {
+	if config.Config.Token == "" && args.authToken == "" {
+		return microerror.Mask(errors.NotLoggedInError)
+	}
+
+	return nil
+}
+
+func printValidation(cmd *cobra.Command, positionalArgs []string) {
+	args := defaultArgs(positionalArgs)
+	err := verifyPreconditions(args, positionalArgs)
+	if err == nil {
+		return
+	}
+
+	errors.HandleCommonErrors(err)
+}
+
+func fetchNodePools(args arguments) (models.V5GetNodePoolsResponse, error) {
+	clientV2, err := client.NewWithConfig(flags.CmdAPIEndpoint, flags.CmdToken)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	auxParams := clientV2.DefaultAuxiliaryParams()
+	auxParams.ActivityName = activityName
+
+	response, err := clientV2.GetNodePools(args.clusterID, auxParams)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// sort node pools by ID
+	sort.Slice(response.Payload[:], func(i, j int) bool {
+		return response.Payload[i].ID < response.Payload[j].ID
+	})
+
+	return response.Payload, nil
+
+}
+
+func printResult(cmd *cobra.Command, positionalArgs []string) {
+	args := defaultArgs(positionalArgs)
+	nodePools, err := fetchNodePools(args)
+	if err != nil {
+		errors.HandleCommonErrors(err)
+		client.HandleErrors(err)
+
+		if clientErr, ok := err.(*clienterror.APIError); ok {
+			fmt.Println(color.RedString(clientErr.ErrorMessage))
+			if clientErr.ErrorDetails != "" {
+				fmt.Println(clientErr.ErrorDetails)
+			}
+		} else {
+			fmt.Println(color.RedString("Error: %s", err.Error()))
+		}
+		os.Exit(1)
```

Comment:
Same as for the error handling. Best practise is to not stop programs somewhere. There is at best one place where the program starts and terminates. That is highly beneficial to code maintenance where different people work on the same project.

Reply:
That's something we should then discuss outside of this PR, as it is done this way in every single command.

Reply:
I brought it up every once in a while but nothing happened really. Somebody needs to feel responsible and drive it and I cannot really do more than bring up the topic again.

Reply:
For the record. We talked about this and want to try to drive certain efforts in the scope of special working hours (SWH). I think this is a pretty good idea to realize more software design related standards.